plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/m2e/core/ui/preferences/MavenArchetypesPreferencePage.java

```diff
@@ -0,0 +1,207 @@
+import org.jboss.reddeer.core.condition.ShellWithTextIsAvailable;
+import org.jboss.reddeer.jface.preference.PreferencePage;
+import org.jboss.reddeer.swt.api.Table;
+import org.jboss.reddeer.swt.api.TableItem;
+import org.jboss.reddeer.swt.impl.button.OkButton;
+import org.jboss.reddeer.swt.impl.button.PushButton;
+import org.jboss.reddeer.swt.impl.combo.DefaultCombo;
+import org.jboss.reddeer.swt.impl.table.DefaultTable;
+import org.jboss.reddeer.swt.impl.text.LabeledText;
+
+/**
+ * Class representing "Maven->Archetypes" preference page.
+ * 
+ * @author jkopriva
+ *
+ */
+
+public class MavenArchetypesPreferencePage extends PreferencePage {
+
+	private static final String ADD_LOCAL_CATALOG = "Add Local Catalog...";
+	private static final String ADD_REMOTE_CATALOG = "Add Remote Catalog...";
+	private static final String EDIT_CATALOG = "Edit...";
+	private static final String REMOVE_CATALOG = "Remove";
+	private static final String LOCAL_CATALOG_SHELL = "Local Archetype Catalog";
+	private static final String REMOTE_CATALOG_SHELL = "Remote Archetype Catalog";
+	private static final String CATALOG_DESCRIPTION = "Description:";
+	private static final String VERIFY_BUTTON = "Verify...";
+
+	/**
+	 * Construct the preference page with "Maven" > "Archetypes".
+	 */
+	public MavenArchetypesPreferencePage() {
+		super(new String[] { "Maven", "Archetypes" });
+	}
+
+	/**
+	 * Add new archetypes local catalog to Eclipse.
+	 * 
+	 * @param catalogFile path to the XML catalog file, e.g. archetype-catalog.xml
+	 * @param description description of the local catalog
+	 */
+	public void addLocalCatalog(String catalogFile, String description) {
+		new PushButton(ADD_LOCAL_CATALOG).click();
+		new WaitUntil(new ShellWithTextIsAvailable(LOCAL_CATALOG_SHELL), TimePeriod.NORMAL);
+		new DefaultCombo().setText(catalogFile);
+		new LabeledText(CATALOG_DESCRIPTION).setText(description);
+		new OkButton().click();
+	}
+
+	/**
+	 * Add new archetypes remote catalog to Eclipse. Catalog is verified.
+	 * 
+	 * @param catalogFileURL url to the remote XML catalog file, e.g. archetype-catalog.xml
+	 * @param description description of the local catalog
+	 */
+	public void addRemoteCatalog(String catalogFileURL, String description) {
+		addRemoteCatalog(catalogFileURL, description, true);
+	}
+
+	/**
+	 * Add new archetypes remote catalog to Eclipse. Catalog is verified depends on verify parameter.
+	 * 
+	 * @param catalogFileURL url to the remote XML catalog file, e.g. archetype-catalog.xml
+	 * @param description description of the local catalog
+	 */
+	public void addRemoteCatalog(String catalogFileURL, String description, boolean verify) {
+		new PushButton(ADD_REMOTE_CATALOG).click();
+		new WaitUntil(new ShellWithTextIsAvailable(REMOTE_CATALOG_SHELL), TimePeriod.NORMAL);
+		new DefaultCombo().setText(catalogFileURL);
+		new LabeledText(CATALOG_DESCRIPTION).setText(description);
+		if (verify) {
+			new PushButton(VERIFY_BUTTON).click();
```

Comment:
What's the purpose of this line? I don't see any difference if the url is correct or incorrect. It just clicks on the Verify button and then continues with clicking on OK.

It would be nice to preserve the verification result. For example you can return it as String result.